src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -0,0 +1,217 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esintgtest;
+
+
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.stream.IntStream;
+
+import static com.amazon.opendistroforelasticsearch.sql.esintgtest.TestsConstants.TEST_INDEX_ACCOUNT;
+import static com.amazon.opendistroforelasticsearch.sql.util.MatcherUtils.hitAny;
+import static com.amazon.opendistroforelasticsearch.sql.util.MatcherUtils.kvDouble;
+import static com.amazon.opendistroforelasticsearch.sql.util.MatcherUtils.kvString;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.equalTo;
+
+
+/**
+ * Created by allwefantasy on 8/25/16.
+ */
+public class SQLFunctionsIT extends SQLIntegTestCase {
+
+    @Override
+    protected void init() throws Exception {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    @Test
+    public void functionFieldAliasAndGroupByAlias() throws Exception {
+        String query = "SELECT " +
+                "floor(substring(address,0,3)*20) as key," +
+                "sum(age) cvalue FROM " + TEST_INDEX_ACCOUNT + "/account where address is not null " +
+                "group by key order by cvalue desc limit 10  ";
+        final JSONObject result = executeQuery(query);
+
+
+        IntStream.rangeClosed(0, 9).forEach(i -> {
+                    Assert.assertNotNull(result.query(String.format("/aggregations/key/buckets/%d/key", i)));
+                    Assert.assertNotNull(result.query(String.format("/aggregations/key/buckets/%d/cvalue/value", i)));
+                }
+        );
+    }
+
+    /**
+     * Commented out the following test as alias for normal fields is no longer stored in SearchHit's fields object
+     */
+//    @Test
+//    public void normalFieldAlias() throws Exception {
```

Comment:
I prefer marking the test with `@Ignore` if we will ever need it, if not let's just remove, included any commented out code, unless there is a critical reason.

We could potentially update the mappings of indices to enforce storing fields,  but it's not urgent, especially if very few testcases are affected.

Reply:
Done.
Create the issue. https://github.com/opendistro-for-elasticsearch/sql/issues/59